packages/fast-components-react-msft/src/button/button.tsx

```diff
@@ -40,23 +37,22 @@ class Button extends Foundation<IButtonHandledProps & IManagedClasses<IMSFTButto
      * Generates class names
      */
     protected generateClassNames(): string {
-        let classLocation: string;
-
-        if (this.props.primary) {
-            classLocation = "managedClasses.button_primary";
-        } else if (this.props.outline) {
-            classLocation = "managedClasses.button_outline";
-        } else if (this.props.lightweight) {
-            classLocation = "managedClasses.button_lightweight";
-        } else if (this.props.justified) {
-            classLocation = "managedClasses.button_justified";
+        switch (this.props.appearance) {
+            case ButtonAppearance.primary:
+                return super.generateClassNames(get(this.props, "managedClasses.button_primary"));
+            case ButtonAppearance.outline:
+                return super.generateClassNames(get(this.props, "managedClasses.button_outline"));
+            case ButtonAppearance.lightweight:
+                return super.generateClassNames(get(this.props, "managedClasses.button_lightweight"));
+            case ButtonAppearance.justified:
+                return super.generateClassNames(get(this.props, "managedClasses.button_justified"));
+            default:
+                return null;
```

Comment:
Can we remove some of the duplicated code here and only return if we have classNames? Perhaps:

```
protected generateClassNames(): string {
    let className: string;

    if (!this.props.appearance) {
        return;
    }

    switch (this.props.appearance) {
        case ButtonAppearance.primary:
            className = "managedClasses.button_primary";
            break;
        case ButtonAppearance.outline:
            className = "managedClasses.button_outline";
            break;
        case ButtonAppearance.lightweight:
            className = "managedClasses.button_lightweight";
            break;
        case ButtonAppearance.justified:
            className = "managedClasses.button_justified";
            break;
    }

    return super.generateClassNames(get(this.props, className));
}
```

@janechu, thoughts?

Reply:
We can avoid the assigning of a variable but I agree we need the if statement.